src/code/views/select-interactive-state-dialog-view.tsx

```diff
@@ -0,0 +1,160 @@
+  }
+
+  handleSelect() {
+    this.props.onSelect(this.props.version.interactiveState)
+    this.props.close?.()
+  }
+
+  handleTogglePreview() {
+    this.setState((prev) => {
+      const preview = !prev.preview
+      this.props.onPreview(preview)
+      return {preview}
+    })
+  }
+
+  getPreviewUrl(interactiveState: any) {
+    // TODO: get preview url from external_report_url
+    return "https://example.com/"
+  }
+
+  UNSAFE_componentWillReceiveProps(props: SelectInteractiveStateVersionProps) {
+    // hide the preview if the overlay text is clicked in the parent
+    if (!props.showingOverlay && this.state.preview) {
+      this.setState({preview: false})
+    }
+  }
+
+  render() {
+    const {preview} = this.state
+    const {version} = this.props
+    const updatedAt = (new Date(version.updatedAt)).toLocaleString()
+    const previewClassName = `preview${preview ? ' preview-active' : ''}`
+
+    return (
+      <div className='version-info'>
+        <div className='dialog-button' onClick={this.handleSelect}>
+          {tr('~DIALOG.SELECT_INTERACTIVE_STATE.USE_THIS_VERSION')}
+        </div>
+        <div className={previewClassName} onClick={this.handleTogglePreview}>
+          <div className='iframe-wrapper'>
+            <iframe className='preview-iframe' src={this.getPreviewUrl(version)}></iframe>
+          </div>
+        </div>
+        <div className='center preview-label' onClick={this.handleTogglePreview}>
+          {tr('~DIALOG.SELECT_INTERACTIVE_STATE.CLICK_TO_PREVIEW')}
+        </div>
+        <table className='version-desc'>
+          <tbody>
+            <tr>
+              <th>{tr('~DIALOG.SELECT_INTERACTIVE_STATE.UPDATED_AT')}</th>
+              <td>{updatedAt}</td>
+            </tr>
+            <tr>
+              <th>{tr('~DIALOG.SELECT_INTERACTIVE_STATE.PAGE')}</th>
+              <td>
+                <span>{version.pageNumber}</span>
+                <span>{version.pageName ? ` - ${version.pageName}` : ""}</span>
+              </td>
+            </tr>
+            <tr>
+              <th>{tr('~DIALOG.SELECT_INTERACTIVE_STATE.ACTIVITY')}</th>
+              <td>{version.activityName}</td>
+            </tr>
+            </tbody>
+        </table>
+      </div>
+    )
+  }
+}
+
+export interface SelectInteractiveStateDialogProps {
+  state1: IInteractiveStateProps<{}>
+  state2: IInteractiveStateProps<{}>
+  interactiveStateAvailable: boolean,
+  onSelect?: SelectInteractiveStateCallback
+  close?: () => void
+}
+export type SelectInteractiveStateCallback = (selected: {}) => void
+
+interface SelectInteractiveStateDialogState {
+  showOverlay: boolean
+}
+
+export default class SelectInteractiveStateDialog extends React.Component<SelectInteractiveStateDialogProps, SelectInteractiveStateDialogState> {
+
+  constructor(props: SelectInteractiveStateDialogProps) {
+    super(props)
+
+    this.handleOnPreview = this.handleOnPreview.bind(this)
+    this.handleHideOverlay = this.handleHideOverlay.bind(this)
```

Comment:
Given that these methods are defined as function expressions I don't think the binding should be necessary here.